examples/applications/object_detection/retina_net/basic/pascal_voc/train.py

```diff
@@ -0,0 +1,341 @@
+        )
+        boxes = boxes.to_tensor(default_value=-1)
+        plotted_images = tf.image.draw_bounding_boxes(images, boxes[..., :4], color)
+        plt.subplot(9 // 3, 9 // 3, i + 1)
+        plt.imshow(plotted_images[0].numpy().astype("uint8"))
+        plt.axis("off")
+    plt.show()
+
+
+visualize_dataset(dataset, bounding_box_format="xywh")
+
+"""
+Looks like everything is structured as expected.  Now we can move on to constructing our
+data augmentation pipeline.
+"""
+
+"""
+## Data augmentation
+
+One of the most labor-intensive tasks when constructing object detection pipeliens is
+data augmentation.  Image augmentation techniques must be aware of the underlying
+bounding boxes, and must update them accordingly.
+
+Luckily, KerasCV natively supports bounding box augmentation with its extensive library
+of [data augmentation layers](https://keras.io/api/keras_cv/layers/preprocessing/).
+The code below loads the Pascal VOC dataset, and performs on-the-fly bounding box
+friendly data augmentation inside of a `tf.data` pipeline.
+"""
+
+# train_ds is batched as a (images, bounding_boxes) tuple
+# bounding_boxes are ragged
+train_ds = load_pascal_voc(
+    bounding_box_format="xywh", split="train", batch_size=FLAGS.batch_size
+)
+val_ds = load_pascal_voc(
+    bounding_box_format="xywh", split="validation", batch_size=FLAGS.batch_size
+)
+
+augmentation_layers = [
+    # keras_cv.layers.RandomShear(x_factor=0.1, bounding_box_format='xywh'),
+    # TODO(lukewood): add color jitter and others
+]
+
+
+def augment(sample):
+    for layer in augmentation_layers:
+        sample = layer(sample)
+    return sample
+
+
+train_ds = train_ds.map(augment, num_parallel_calls=tf.data.AUTOTUNE)
+visualize_dataset(train_ds, bounding_box_format="xywh")
+
+"""
+Great!  We now have a bounding box friendly augmentation pipeline.
+
+Next, let's unpackage our inputs from the preprocessing dictionary, and prepare to feed
+the inputs into our model.
+"""
+
+
+def unpackage_dict(inputs):
+    return inputs["images"], inputs["bounding_boxes"]
+
+
+train_ds = train_ds.map(unpackage_dict, num_parallel_calls=tf.data.AUTOTUNE)
+val_ds = val_ds.map(unpackage_dict, num_parallel_calls=tf.data.AUTOTUNE)
+
+train_ds = train_ds.prefetch(tf.data.AUTOTUNE)
+val_ds = val_ds.prefetch(tf.data.AUTOTUNE)
+
+"""
+Our data pipeline is now complete.  We can now move on to model creation and training.
+"""
+
+"""
+## Model creation
+
+We'll use the KerasCV API to construct a RetinaNet model.  In this tutorial we use
+a pretrained ResNet50 backbone using weights.  In order to perform fine-tuning, we
+freeze the backbone before training.  When `include_rescaling=True` is set, inputs to
+the model are expected to be in the range `[0, 255]`.
+"""
+
+model = keras_cv.applications.RetinaNet(
+    classes=20,
+    bounding_box_format="xywh",
+    backbone="resnet50",
+    backbone_weights="imagenet",
+    include_rescaling=True,
+)
+model.backbone.trainable = False
```

Comment:
Should this happen by default with our API?

Maybe this is a question for Martin, but I would think most users would want the backbone to be untrainable by default.

Reply:
I don't think so.  This would fall in the category of surprising behavior, it just depends on your use case.  People do basically every permutation of every configuration of what is trainable and what isn't in OD.